src/hdr_histogram.c

```diff
@@ -409,64 +467,81 @@ int hdr_init(
         int64_t lowest_discernible_value,
         int64_t highest_trackable_value,
         int significant_figures,
-        struct hdr_histogram** result)
+        struct hdr_histogram* histogram)
 {
-    int64_t* counts;
     struct hdr_histogram_bucket_config cfg;
-    struct hdr_histogram* histogram;
 
     int r = hdr_calculate_bucket_config(lowest_discernible_value, highest_trackable_value, significant_figures, &cfg);
     if (r)
     {
         return r;
     }
 
-    counts = (int64_t*) hdr_calloc((size_t) cfg.counts_len, sizeof(int64_t));
-    if (!counts)
+    if (!histogram)
     {
         return ENOMEM;
     }
 
-    histogram = (struct hdr_histogram*) hdr_calloc(1, sizeof(struct hdr_histogram));
+    hdr_init_preallocated(histogram, &cfg);
+
+    return 0;
+}
+
+int hdr_init_pgss(
+        int64_t lowest_discernible_value,
+        int64_t highest_trackable_value,
+        int yb_bucket_factor,
+        struct hdr_histogram* histogram)
```

Comment:
I noticed that there is Postgres-specific code directly integrated into the library. I'm wondering if it would be better to move the entire codebase into the YugabyteDB repository, since it seems to have a Postgres-specific focus. Alternatively, we could extract the Postgres-specific code into its own file, which would make merging changes from upstream easier in the future. What are your thoughts on this?

Reply:
I think the option of extracting the postgres-specific code into its own file would be useful. I'm not as favorable towards bringing it entirely into YugabyteDB, but if you see future use cases / more intense work on hdr_histogram functionality that would end up being useful.  Considering the time constraints on this project, I think it's a good matter for later.

Reply:
Fair. Let's leave it up to your discretion.